src/parser.rs

```diff
@@ -154,26 +124,24 @@ fn collapse_chunks(chunks: &[EncodedChunk]) -> Result<Vec<EncodedChunk>, Restore
 
     let mut mismatching = Vec::<(EncodedChunk, String)>::new();
     // Group by chunk id
-    for (_chunk_id, duplicate_chunks) in &chunks.into_iter().group_by(|chunk| chunk.id) {
-        let mut duplicate_chunks_iter = duplicate_chunks.into_iter();
-        let reference_chunk = match duplicate_chunks_iter.next() {
+    for (_chunk_id, mut duplicate_chunks) in &chunks.iter().group_by(|chunk| chunk.id) {
+        let reference_chunk = match duplicate_chunks.next() {
             Some(v) => v,
             None => panic!("Couldn't get 1 item per groupby"),
         };
-        for duplicate_chunk in duplicate_chunks_iter {
+        for duplicate_chunk in duplicate_chunks {
             if duplicate_chunk.payload != reference_chunk.payload {
                 mismatching.push((reference_chunk.clone(), duplicate_chunk.payload.clone()));
             }
         }
         chunks_out.push(reference_chunk.clone());
     }
-    match mismatching.is_empty() {
-        true => return Ok(chunks_out.to_vec()),
-        false => {
-            return Err(RestoreError::MismatchingDuplicateChunk {
-                mismatching_chunks: mismatching,
-            })
-        }
+    if mismatching.is_empty() {
+        Ok(chunks_out)
+    } else {
+        Err(RestoreError::MismatchingDuplicateChunk {
+            mismatching_chunks: mismatching,
+        })
```

Comment:
That's clearer indeed, I just worried that every time I use `if (a) {x} else {y}` I am writing imperative code that brings my family shame, and every time I use `match` I am a proud rustacean.

This is honestly one of the biggest trip-ups I'm working with right now, how options/error/result get propagated/created/popped around because I don't know the syntax properly. Like "I have a Option<Result<(), SomeEnum>>, given I know it's Some not None, how do I pop the Some-ness" etc.

Luckily it's small potatoes, as far as issues go.

Reply:
>Like "I have a Option<Result<(), SomeEnum>>, given I know it's Some not None, how do I pop the Some-ness"

if you *know* it's `Some` and not `None`, you can 'unwrap' it. but i would question why you have an Option type if you know it's not None?

which bit of code are you referring to exactly?